test/helpers/gather-os-metrics.spec.js

```diff
@@ -0,0 +1,20 @@
+const chai = require('chai');
+
+chai.should();
+
+const gatherOsMetrics = require('../../helpers/gather-os-metrics');
+
+describe('helpers', () => {
+  describe('gather-os-metrics', () => {
+    describe('when invoked', () => {
+      it('then ...', () => {
+        const span = { os: [], responses: [] };
+
+        gatherOsMetrics({}, span);
+
+        // todo: not sure what should I test, maybe the resulted span structure?
+        // todo: also this component has got some internal timing events?
```

Comment:
good idea @soyuka re process.pid and mocking pidusage; also can you confirm that the component uses some kind of settimeout internally?